### ISSUE TITLE
Merge matching structs between circuit input builder and evm circuit witness (where possible)

### ISSUE THREAD

**Issue by ed255** [privacy-scaling-explorations/zkevm-circuits#528]
The idea of having duplicate structs between circuit input builder and evm witness generation was to allow the development of the circuit input builder in parallel with the circuits while minimizing the changes that one implies to the other. But once things stabilize, I think we can merge those.

Related to https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/484#discussion_r875590637

See the latest discussions upon that:
> Unify the types `Block`,` Call`, `Step`, `Transaction` between these two:
>
>    - zkevm-circuits/src/witness
>    - bus-mapping/src/circuit_input_builder

Also see: 
> Remove duplicated fields in eth_block -> https://github.com/appliedzkp/zkevm-circuits/blob/feat%2Fannotate_evm_cols/bus-mapping/src/circuit_input_builder/block.rs#L54-L57
> (Work on refactoring and have just 2 blocks, 1 for RPC and the other for general purpose)

**Comment by DreamWuGit** [comment]
Really good idea to reduce redundant  structs !!!

**Comment by ChihChengLiang** [comment]
I can take this one.

**Comment by ed255** [comment]
There are still some structs that can be de-duplicated after the PR https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1378 @ChihChengLiang will open a new issue to track those.
